### ISSUE TITLE
SegmentIntegration.flushScheduler not shutdown in Analytics.shutdown

### ISSUE THREAD

**Issue by charlesdurham** [segmentio/analytics-android#616]
Version: 3.4.1

When we switch from our prod to dev environments we switch analytic instances (different API keys) and call shutdown on the old one. 

Doing this causes:

```
    java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@58269de rejected from com.segment.analytics.internal.Utils$AnalyticsNetworkExecutorService@47ce7bf[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2085)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
        at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118)
        at com.segment.analytics.SegmentIntegration.submitFlush(SegmentIntegration.java:314)
        at com.segment.analytics.SegmentIntegration$SegmentDispatcherHandler.handleMessage(SegmentIntegration.java:509)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
```

From what I can tell the error is caused because `analytics.shutdown` shuts down the network executor used to flush in `SegmentIntegration` but not the `flushScheduler` which schedules the work periodically. 

This results in trying to perform a flush after the `networkExecutor` is shutdown.

**Comment by f2prateek** [comment]
We haven't seen this issue before, but I've added this to our [internal JIRA](https://segment.atlassian.net/browse/LIB-1168), I would suggest reaching out to our support team so they can keep you updated with the progress.

**Comment by jlmcdonnell** [comment]
If #628 was meant to fix this, I'm still seeing this crash in `4.5.0-beta.0`

**Comment by prayansh** [comment]
@charlesdurham @jlmcdonnell fix should have been released as of `4.5.0-beta.1`.
Gonna close this for now. Reopen if issue still persists
